### DISCUSSION TITLE
Cant build Heltec compile errors M5

### DISCUSSION THREAD

**Discussion by imCharlieB** [PlummersSoftwareLLC/NightDriverStrip#38] [Q&A]
I got  my heltec esp32 in and was trying to build it and i am getting some errors.. first i can build and deploy the demo ( my leds dont seem to work and the web server is not responding ) So after the demo worked kind of lol i turned on the display and that works i get an IP and serial out doesnt show any errors other than a C1 Failed GET for /

so onto trying the heltec and then build i ran into this error Compiling `Compiling .pio\build\heltec\lib207\IRremoteESP8266\ir_EliteScreens.cpp.o
In file included from src/effects.cpp:43:0:
include/effects/spectrumeffects.h: In member function 'void VUMeterEffect::DrawVUMeter(int)':
include/effects/spectrumeffects.h:122:13: error: 'M5' was not declared in this scope
             M5.Lcd.fillRect(0, 0, M5.Lcd.width(), 10, BLACK16);
             ^
In file included from src/effects.cpp:43:0:
include/effects/spectrumeffects.h: In member function 'virtual void SpectrumAnalyzerEffect::Draw()':
include/effects/spectrumeffects.h:353:17: error: 'M5' was not declared in this scope
                 M5.Lcd.fillScreen(BLACK16);
                 ^
Compiling .pio\build\heltec\lib207\IRremoteESP8266\ir_Epson.cpp.o`  any ideas?

**Comment by imCharlieB** [comment]
welllll that was silly of me... i changed the platforio.ini for the heltec to build_flags   = -DDEMO=1 and it runs fine now its not trying to run the spectrum

**Reply by BeldrothTheGold** [reply]
Indeed, the spectrum example is intended to run on an M5Stick Plus. It uses the built-in microphone. 
I will submit a PR to change the build-flags for the included heltec example.

**Comment by robertlipe** [comment]
@imCharlieB , it seems you answered your own question in https://github.com/PlummersSoftwareLLC/NightDriverStrip/discussions/38#discussioncomment-1460757. In the years since this was filed, there's now some amount of automated tests to help keep the trunk running on common targets.

Please consider marking this as "answered" unless you thin kthere's something left to do here all these years later.

Thanx!
